### DISCUSSION TITLE
I can't download a long video on YouTube until the end.

### DISCUSSION THREAD

**Discussion by pu22222** [aclap-dev/video-downloadhelper#53] [Help and Q&A] [answered]
### Your question

I am unable to download the YouTube video at the URL below.
It ends when I have downloaded about 10 to 16 MB.
For short videos of about 10 minutes, there is no problem, but for longer videos, the download often ends in the middle of downloading.

https://www.youtube.com/watch?v=IVD_CQXc5pY

version: 8.2.0.18a1
channel: beta
build date: 2024-02-01
build options: linuxlic: false, noyt: false, browser: firefox
lang: ja
license: accepted
key (only 16 first characters): 34bea62fc65b4126
platform: x86-64 win
UA: Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:122.0) Gecko/20100101 Firefox/122.0
coapp: yes. Info:
{
  "id": "net.downloadhelper.coapp",
  "name": "VdhCoApp",
  "version": "2.0.14",
  "binary": "C:\\Program Files\\DownloadHelper CoApp\\vdhcoapp.exe",
  "displayName": "VdhCoApp",
  "description": "Video DownloadHelper companion app",
  "target": {
    "os": "windows",
    "arch": "x86_64",
    "node": "18"
  },
  "home": "C:\\Users\\samia",
  "converterBinary": "C:\\Program Files\\DownloadHelper CoApp\\ffmpeg.exe",
  "converterBase": "ffmpeg",
  "converterBaseVersion": "63b936e978"
}

**Comment by GrampaWildWilly** [comment]
For the record, I am running Windows 11 Pro 64-bit version 23H2 build 22631.3085, Firefox 122.0.1 64-bit, licensed VDH 8.2.0.19a1 beta, CoApp 2.0.17.

When I went to your YouTube page, it looked like this:

![#01](https://github.com/aclap-dev/video-downloadhelper/assets/75588964/892b2d88-aa6e-4817-9e34-0da7b26464b0)

![#02](https://github.com/aclap-dev/video-downloadhelper/assets/75588964/aa163189-f12c-4be9-8177-b1cab88af465)

Here are the resulting files VDH downloaded:

![#03](https://github.com/aclap-dev/video-downloadhelper/assets/75588964/4145398c-70e5-4cdb-8497-a95b29d0a1f6)

And here's more results:

![#08](https://github.com/aclap-dev/video-downloadhelper/assets/75588964/b51a275f-14fb-4028-93f9-76d2c2931745)

I know.  Unless you're an ant, you can't read that.  No problem.  Just click the image.  That opens it in a separate browser window, where you can read it, even magnify it if necessary.

I was a bit perturbed by the frame rate of just 6fps.  But then I went back to the YouTube page & saw that it's a radio station.  When I played the mp4 & the mkv in VLC, it turned out that the entire video track is just a fixed image.  You don't get video from a radio station.  So this now makes sense.  I didn't sit & listen to this, just sampled it at intervals.  I had the fixed image with something on the audio at each sample point.  I don't speak Japanese so you might ask how do I know this is Japanese?  I copy/pasted the video title from the page into Google Translate & it told me it was Japanese.  Despite that, I'm sure this is an excellent radio station.  Since I played about the last 20 seconds (I always play the very end of something to make sure it's all there), I was able to hear a bit of Herb Alpert & the Tijuana Brass.  Clearly, this DJ has highly refined good taste.

So there you have it.  No problems whatsoever.  You might want to get on the same versions of VDH & CoApp.

**Comment by paulrouget** [comment]
Can you try this:

- Download the latest CoApp: https://github.com/aclap-dev/vdhcoapp/releases/latest/download/vdhcoapp-windows-x86_64-installer.exe
- close firefox
- install the CoApp
- open firefox
- go to the youtube page
- take a screenshot of the detected videos
- **choose the fist entry**

Share the screenshot and let me know how it goes.

**Comment by pu22222** [comment]
Thanks for the advice.
However, I have updated CoApp and it has not improved.
Until last month I was able to save even the longest videos without any problems, but this problem has been happening since I updated this month.

![Mozilla Firefox 20240209 194543](https://github.com/aclap-dev/video-downloadhelper/assets/159499328/2c8d17da-7e7e-4427-a350-f81072fca3ae)
![Explorer 20240209 194851](https://github.com/aclap-dev/video-downloadhelper/assets/159499328/25b8b9fb-0e3d-4861-8e65-2a8d2d92ba38)

version: 8.2.0.19a1
channel: beta
build date: 2024-02-08
build options: linuxlic: false, noyt: false, browser: firefox
lang: ja
license: accepted
key (only 16 first characters): 34bea62fc65b4126
platform: x86-64 win
UA: Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:122.0) Gecko/20100101 Firefox/122.0
coapp: yes. Info:
{
  "id": "net.downloadhelper.coapp",
  "name": "VdhCoApp",
  "version": "2.0.17",
  "binary": "C:\\Program Files\\DownloadHelper CoApp\\vdhcoapp.exe",
  "displayName": "VdhCoApp",
  "description": "Video DownloadHelper companion app",
  "target": {
    "os": "windows",
    "arch": "x86_64",
    "node": "18"
  },
  "home": "C:\\Users\\samia",
  "converterBinary": "C:\\Program Files\\DownloadHelper CoApp\\ffmpeg.exe",
  "converterBase": "ffmpeg",
  "converterBaseVersion": "63b936e978"
}

**Comment by pu22222** [comment]
I tried it on another video.

https://www.youtube.com/watch?v=iwrFGjqXby8
No problem.

https://www.youtube.com/watch?v=ZZMAV7K1zDc
Longer videos still end in the middle of the video.

![20240211 224733 Mozilla Firefox](https://github.com/aclap-dev/video-downloadhelper/assets/159499328/2eb56611-d7c7-479c-a762-9c19f531b57c)

![20240211 224804 Mozilla Firefox](https://github.com/aclap-dev/video-downloadhelper/assets/159499328/ca9a1461-7134-4fc6-8e22-74dbf6bb1b24)

![20240211 224940 Mozilla Firefox](https://github.com/aclap-dev/video-downloadhelper/assets/159499328/df189109-e0db-477f-a1a6-b94c7cacd806)

![20240211 225044 Explorer](https://github.com/aclap-dev/video-downloadhelper/assets/159499328/db8beed1-be96-4ed8-98da-8a02f33023e3)

**Reply by BrokenEq** [reply]
I too have this problem, short videos download insanely fast, but longer videos bog down and usually stop somewhere in the middle, even when I'm only downloading a single video... this behavior didn't use to happen.  I even bought a registration today to see if that helped fix it, but still no luck...

**Reply by paulrouget** [reply]
Please share the URL of the failing video.

**Reply by BrokenEq** [reply]
Here is the sample url I was trying, it's just a video I know is rather long / large: https://www.youtube.com/watch?v=xRpWI4kRzic&

**Reply by BrokenEq** [reply]
Here's also an example of the crawling speeds I'm talking about, whereas if it's a shorter video these are in MB/s:
![image](https://github.com/aclap-dev/video-downloadhelper/assets/159967844/8702b5bf-2b20-459b-bbdd-6f7d28b0d242)

**Reply by paulrouget** [reply]
I get the same speed. Sadly, it's often youtube that caps the download speed. Not something we can do anything about.

Something you can try: Settings > More > Video Qualities

Uncheck "Ignore low quality videos".

Now, in youtube, you'll see an extra option to download from. See if that changes anything.

**Reply by BrokenEq** [reply]
Unfortunately no, I still have extremely slow speeds when using the extension and if I download multiple things at the same time it halts about halfway through on most of them.

**Reply by paulrouget** [reply]
> things

I need details, otherwise I can't help.

- what are the URLs of these videos
- how many you download in parallel
- when it halts, does it show errors, or it just ends
- are the video playable

**Reply by BrokenEq** [reply]
Understand, so the difficulty of sharing the urls is they are behind a paysite, so even if I shared them you wouldn't be able to access them.  

As for the other information:

**2) how many you download in parallel** - I download between 6 and 9 usually, but I noticed this issue even when only downloading 1 or 2 at the same time.

**3) when it halts, does it show errors, or it just ends** - It just stops downloading, no errors or anything the downloads just halt

**4) are the video playable** - For the ones that actually finish yes, they are playable, but the ones that halt, not as they are incomplete

**Reply by BrokenEq** [reply]
Here's some more info, I'm downloading a single file, when I go to download it shows very high speeds for the download: 
![image](https://github.com/aclap-dev/video-downloadhelper/assets/159967844/8e5ed2b5-f2cb-4aa1-b95e-b8dba13b4b48)

But then when I actually start downloading it:
![image](https://github.com/aclap-dev/video-downloadhelper/assets/159967844/7eb8cfb3-9c9b-4474-bbf2-ffa54241eaee)

**Reply by paulrouget** [reply]
Thank you, this is useful. I'll look it up.

**Reply by pu22222** [reply]
>Uncheck "Ignore low quality videos".

This setup had already been tried.
I don't know if this setting has an effect, but when I see the MP4 choice further down in the attached image, where there are 3 choices each for MP4 and MKV, only this choice can be saved to the end.

The differences from the other choices are,
1. the speed is much slower as it was last month.
2. It is not possible to choose whether the resolution is 1920x1200 or 256x144, only that it will save successfully in either case.

![Mozilla Firefox 20240216 92955](https://github.com/aclap-dev/video-downloadhelper/assets/159499328/e63d7216-8499-4836-99a5-19248c6ab59d)

**Reply by pu22222** [reply]
These URLs also failed to download.

https://www.youtube.com/watch?v=w_787odHuQ0
https://www.youtube.com/watch?v=Pv-ZJND9VzE
https://www.youtube.com/watch?v=4uy7OWmDXeI

**Reply by paulrouget** [reply]
I could also try this one as https://www.youtube.com/watch?v=4uy7OWmDXeI

and it's working here. The other ones are not available in my country.

I think it might have to do with region specific stream mechanisms?

**Reply by pu22222** [reply]
I recently got a sub-laptop and tried to see if I could save a video that failed to save in the new environment.
However, my license was locked, probably because I entered the license key in a location that was not my home.
I am currently in the process of reissuing it.

Since I was so curious, I tried to save it without the license, and of course the QR code is included, but I succeeded in saving it with the same slow download speed as before.
This is both on my sub-laptop and on my main computer where it had been failing.

I will try again as soon as the license is reissued.

**Reply by GrampaWildWilly** [reply]
Don't wait.  Your license will not be reissued.  You can unlock it yourself.  On the first page of VDH settings, in the bottom section for the license (click it to expand it), these is a button for Recheck license.  Click that.  That is supposed to send you an E-mail.  Follow the instructions in the E-mail.  You should then click the Recheck license button again on all your computers.  A VDH license is for life & covers all the computers you own now as well as any computers you may own in the future.  You may simply hit this locked license condition occasionally.  But you will be able to fix it yourself as I have described here.  But don't post your license number in public.  EVER.  And don't share you license with your family & friends.  Make them buy their own licenses.

**Reply by pu22222** [reply]
I am definitely following that procedure, thank you.
I just have not been able to get a reply email very quickly.
I am using DeepL to translate it, but it may not be an accurate representation of my intent.

**Reply by GrampaWildWilly** [reply]
Look in the Wiki (link at the top of this page).  Look in the section titled "I Need Help."  A section of that discusses license issues.  Get more information there.

**Reply by pu22222** [reply]
I am not asking about the license.
The main issue is that I was able to download successfully without a license.

**Comment by paulrouget** [comment]
Do you download the MP4 or the MKV?

**Reply by pu22222** [reply]
Yes.

**Reply by paulrouget** [reply]
Which one, MP4 or MKV?

**Reply by pu22222** [reply]
Both MP4 and MKV are no good, but I think MKV is more prone to occur.
I prefer to save in MKV.

**Reply by BrokenEq** [reply]
For me I only do mp4... so defer to @pu22222  for mkv knowledge.

**Comment by GrampaWildWilly** [comment]
The thought has crossed my mind that the ISP is preventing the downloads from completing.  Maybe the ISP is imposing some sort of volume limit on downloads.  Just a thought.  Could be totally off base.

**Reply by pu22222** [reply]
I have considered that possibility, as well as other issues with my router, etc.
However, I have downloaded about 1.6GB of previously saved MKVs from Google Drive and it completes without any problems at about 900Mbps.
So it is hard to imagine some kind of limitation or hardware failure.

**Reply by BrokenEq** [reply]
Yeah, I thought that as well, but considering I get this only with using this add on and in both Chrome and Firefox and ONLY when downloading large files (small files download really fast) and considering normal downloads (outside the extension) are also very fast, I also consider it unlikely.

**Reply by paulrouget** [reply]
@pu22222 try this: https://github.com/aclap-dev/video-downloadhelper/discussions/53#discussioncomment-8474640

**Reply by pu22222** [reply]
Sorry for the late reply.
I recently had trouble recognizing videos on YouTube and remembered about this place.
I tried “Uncheck ‘Ignore low quality videos’” but it didn't help.

After installing 9.3.0.0, the videos are now recognized, but the situation has not improved since a year ago.

https://github.com/aclap-dev/video-downloadhelper/discussions/53#discussioncomment-8616672
Is it not possible to support “region specific stream mechanisms”?

**Reply by mateusloubach** [reply]
dropping in on the conversation here, because I'm currently dealing with the same issue. - It doesn't seem to be hardware issues, or internet. I don't know how far you guys have gone to solve this, but it seems to be something hard-coded into the system to prevent a high amount of downloads.

My case is that I've downloaded many videos and it just started to take a longer time to download (although I've a 700mpbs and am the sole user on this line) and it just finishes at around 90%.Tried going premium, but they're having issues with the stripe page. (it won't complete the order).

**Reply by pu22222** [reply]
https://www.youtube.com/watch?v=tGut3aI3zkg
https://github.com/user-attachments/assets/c960943a-405c-465b-a1f7-6b71f9395c07

I took a video of the behavior when saving fails.
We interrupt the video when we have saved the first 4 minutes of the video.
Sometimes it works by changing the format in which it is saved, and sometimes it makes no sense at all.
Sometimes it works if you restart Firefox and save immediately.
Sometimes if you wait a few days and try it, it saves as if nothing happened.
However, I have been trying various methods for a year and have not found one that always works.

**Comment by pu22222** [comment, marked as answer]
I finally found a solution.
By setting bandwidth restrictions on my router, I was able to download videos that I couldn't finish downloading before.
My internet connection speed is 1 Gbps, but 100 Mbps seems to be sufficient.
It's not a fundamental solution, but I'm glad I was able to achieve my goal.
